psycop/common/sequence_models/dataset.py

```diff
@@ -16,3 +16,16 @@ def __len__(self) -> int:
 
     def __getitem__(self, idx: int) -> Patient:
         return self.patients[idx]
+
+
+class PatientDatasetWithLabels(Dataset):
+    def __init__(self, patients: list[Patient], labels: list[int]) -> None:
+        self.patients: list[Patient] = patients
+        self.labels: list[int] = labels
+        assert len(self.patients) == len(self.labels)
+
+    def __len__(self) -> int:
+        return len(self.patients)
+
+    def __getitem__(self, idx: int) -> tuple[Patient, int]:
+        return self.patients[idx], self.labels[idx]
```

Comment:
I'll rewrite this to use `PredictionTime` if you'd like 👍

Reply:
moved to issue #263